accepted/00102-upgrade-to-salt-3008.md

```diff
@@ -0,0 +1,124 @@
+- Start Date: 2024-10-31
+
+# Summary
+[summary]: #summary
+
+This RFC describes the preparation and strategy to upgrade Salt to the upcoming 3008.x release (LTS).
+
+# Motivation
+[motivation]: #motivation
+
+The upstream Salt project is aiming the future 3008.x release (LTS) for Autumn 2024. As we want Uyuni and SUSE Manager to align to the latest LTS version from Salt project, we need to prepare the strategy to drive this upgrade.
+
+Particularly, this new 3008.x LTS release is the first version after [the great module migration](https://salt.tips/the-great-salt-module-migration/), and lot of Salt modules have been purged from the main Salt package in favor of [Salt Extensions](https://github.com/salt-extensions), including for example, the `zypperpkg` and `transactional_update` modules.
+
+At this point in time, only few modules have been already migrated to Salt Extensions, a lot of other modules are still waiting at [community-extensions-holding repository](https://github.com/salt-extensions/community-extensions-holding) to be migrated.
+
+The purpose of this RFC is to define:
+- Salt and Salt Extensions packaging strategy.
+- How to customize the Salt package with more extensions.
+- New dependencies required for Salt.
+- How to release Salt 3008.x to SLE.
+- How to maintain Salt 3008.x going forward.
+
+# Detailed design
+[design]: #detailed-design
+
+## Packaging
+
+Due the great module purge, we need to rethink the way we package Salt, as lot of the modules are been dropped and Salt Extensions are stepping in now.
+
+### Builtin Salt Extensions
+
+There are a couple of Salt Extensions that we want to have integrated by default in our main `python311-salt` package. These are:
+- zypperpkg
+- dpkg
+- transactional_update
+- openscap
+- docker / dockerng
+- ...
+
+These are the minimum required extensions to be able to do basic operations in SUSE/openSUSE distributions and also allow basic operations on the context of SUSE Manager clients.
+
+Alternatively, we could reduce this list even more by just providing the `zypperpkg` and `transactional_update` modules and then provide the rest via `/usr/share/susemanager/salt/[_modules,_states,...]` in the Uyuni / SUSE Manager server.
+
+### What about Salt Extensions packages?
+
+Essentially there are two different approaches here:
+
+1. Creating separated packages for each one of the Salt Extensions (potentially hundreds) -> maybe too cumbersome if we go with all of them, and probably not really needed.
+2. Do not package Salt Extensions at all -> builtin extensions + manual customization.
+
+With option 1, we do package and do support all those extensions we decide to have a package for. On the other hand, with option 2, we do only take care of maintaing and supporting the reduced list of builin extensions (actually integrated in the main `python311-salt` package), and then users and customers would need to rely on extending with their desired extensions coming from the community.
```

Comment:
There could be a third option, packaging a curated list of salt extensions. The benefit would be that Uyuni can mirror RPM repos, but not PyPI. Our current extension approach requires internet access or an independent PyPI mirror.

I don't think it's worthwhile to package all extensions, this is a good opportunity to focus our efforts on the Salt extensions that are popular and make sure we can test them as well.